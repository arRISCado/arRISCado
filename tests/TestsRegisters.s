	.option nopic
	.text
	.align	2
	.globl	main
	.type	main, @function
main:
	addi zero, a0, 1
	addi t0, zero, 2
	addi t1, zero, 3
	addi t2, zero, 4
	addi s0, t0, 5
	addi s1, t1, 6
	addi a0, t2, 7
	addi a1, s0, 8
	addi a2, s1, 9
	addi a3, a0, 10
	addi a4, a1, 11
	addi a5, a2, 12
	addi a6, a3, 13
	addi a7, a4, 14
	addi s2, a5, 15
	addi s3, a6, 16
	addi s4, a7, 17
	addi s5, s2, 18
	addi s6, s3, 19
	addi s7, s4, 20
	addi s8, s5, 21
	addi s9, s6, 22
	addi s10, s7, 23
	addi s11, s8, 24
	addi t3, s9, 25
	addi t4, s10, 26
	addi t5, s11, 27
	addi t6, t3, 28

# Expected final state:
# zero: 0
# t4: 126
# t5: 135
# t6: 144
