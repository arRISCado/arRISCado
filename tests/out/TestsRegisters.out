0           x0	zero
x           x1	ra
x           x2	sp
x           x3	gp
x           x4	tp
x           x5	t0
x           x6	t1
x           x7	t2
x           x8	s0
x           x9	s1
x           x10	a0
x           x11	a1
x           x12	a2
x           x13	a3
x           x14	a4
x           x15	a5
x           x16	a6
x           x17	a7
x           x18	s2
x           x19	s3
x           x20	s4
x           x21	s5
x           x22	s6
x           x23	s7
x           x24	s8
x           x25	s9
x           x26	s10
x           x27	s11
x           x28	t3
126         x29	t4
135         x30	t5
144         x31	t6